File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 5/29/2026, 12:01:22 PM
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;
        }
        else{
            String s=Integer.toString(x);
            int left=0,right=s.length()-1;
            while(left<right){
                if(s.charAt(left)!=s.charAt(right)){
                    return false;
                }
                left++;
                right--;
            }
            return true;
        }
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 5/29/2026, 12:01:18 PM
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        str1=str(x)
        return str1==str1[::-1]